src/dll/Systems/ScriptCompilationSystem.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "Hook.hpp"
+#include "ISystem.hpp"
+#include "Paths.hpp"
+#include "PluginBase.hpp"
+
+struct FixedWString
+{
+    uint32_t length;
+    uint32_t maxLength;
+    wchar_t* str;
+};
+
+class ScriptCompilationSystem : public ISystem
+{
+public:
+    ScriptCompilationSystem(const Paths& aPaths);
+
+    ESystemType GetType() final;
+
+    void Startup() final;
+    void Shutdown() final;
+
+    bool Add(std::shared_ptr<PluginBase> aPlugin, const wchar_t* aPath);
+    void SetScriptsBlob(const std::filesystem::path& aPath);
+    const std::filesystem::path& GetScriptsBlob() const;
+    bool HasScriptsBlob() const;
+    FixedWString GetCompilationArgs(const FixedWString& original);
```

Comment:
```suggestion
    bool Add(std::shared_ptr<PluginBase> aPlugin, const wchar_t* aPath);
    
    void SetScriptsBlob(const std::filesystem::path& aPath);
    const std::filesystem::path& GetScriptsBlob() const;
    bool HasScriptsBlob() const;
    
    FixedWString GetCompilationArgs(const FixedWString& aOriginal) const;
```
I like to put a newline between "groups". 🏃